ise.py

```diff
@@ -57,9 +65,78 @@ def __init__(
+
+    @staticmethod
+    def discover_primary_ers_node(self):
+        """
+        Loops through the seed node's knowledge of the ERS nodes.
+        Queries each node if it has the Primary Active Role for ERS
+        Returns the Primary Node ip address.
+        """
+        
+        # Choosing JSON instead of XML
+        self.ise.headers.update(
+            {"ACCEPT": "application/json", "Content-Type": "application/json"}
+        )
+        try:
+            resp = self._request(
+                "{0}/config/node".format(self.url_base), method="get"
+            )
+
+            if resp.status_code == 200:
+                nodes = [
+                    (i["id"],i["link"]["href"])
+                    for i in resp.json()["SearchResult"]["resources"]
+                ]
+                for node in nodes:
+                    try:
+                        node_response = self._request(
+                        "{0}/config/node/{1}".format(self.url_base, node[0]), method="get"
+                        )
+                        if(node_response.json()["Node"]["primaryPapNode"]):
+                            print("Found Primary ERS Node")
+                            return node_response.json()["Node"]["ipAddress"]
+                    except requests.exceptions.RequestException as e:
+                        print("Error: Connection Error to Primary ISE Node, continuing...")
+        except requests.exceptions.RequestException as e:
+            print("Error: The primary node is not responding, trying secondary node.")
+            # If we get here, the first node failed to respond, we try the second node.
+
+            try:
+                secondary_url_base = "{0}://{1}:9060/ers".format(self.protocol, self.ise_node_2)
+                resp = self._request(
+                "{0}/config/node".format(secondary_url_base), method="get"
+                )
+
+                if resp.status_code == 200:
+                    nodes = [
+                    (i["id"],i["link"]["href"])
+                    for i in resp.json()["SearchResult"]["resources"]
+                ]
+                for node in nodes:
+                    # Because all the nodes are known, they may timeout here also.
+                    try:
+                        node_response = self._request(
+                        "{0}/config/node/{1}".format(secondary_url_base, node[0]), method="get"
+                        )
+                        if(node_response.json()["Node"]["primaryPapNode"]):
+                            print("Found Primary ERS Node")
+                            return node_response.json()["Node"]["ipAddress"]
+                    except requests.exceptions.RequestException as e:
+                        print("Error: Connection Error to Secondary ISE Node.")
+                        raise
+            except requests.exceptions.RequestException as e:
+                print("Error: Connection Error to Secondary ISE Node.")
```

Comment:
All prints should be converted into logging statements. 

```py
import logging

log = logging.getLogger('ise')
log.debug(…)
log.warning(…)

```

Reply:
Added logging to replace prints.